### PR TITLE
mcs add bucket event api using public mc S3Client struct

### PR DESCRIPTION
**Pull request by cesnietor**
API creates a bucket notification event:
`POST` `/api/v1/buckets/{bucket_name}/events`
Payload example:
```
{
	"configuration":
		{
            "arn": "arn:minio:sqs::test:postgresql",
            "events": ["put", "get"],
            "prefix": "photos/",
            "suffix": ".jpg"
        }
}
```
or
```
{
	"configuration":
		{
            "arn": "arn:minio:sqs::test:postgresql"
        },
    "ignoreExisting": true
}
```
Note: wait until https://github.com/minio/mc/pull/3139 is merged which makes `S3Client` public for mc

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,9 +12,11 @@ require (
...
-	github.com/minio/mc v0.0.0-20200401220942-e05f02d9f459
+	github.com/minio/mc v0.0.0-20200402152133-51ef254a411d
 	github.com/minio/minio v0.0.0-20200327214830-6f992134a25f
-	github.com/minio/minio-go/v6 v6.0.51-0.20200319192131-097caa7760c7
+	github.com/minio/minio-go/v6 v6.0.51-0.20200401083717-eadbcae2a0e6
 	github.com/stretchr/testify v1.5.1
 	golang.org/x/net v0.0.0-20200301022130-244492dfa37a
 )
+
+replace github.com/minio/mc => github.com/cesnietor/mc v0.0.0-20200403000513-170fa50e4d5b
```
Comment (dvaldivia) [inline go.mod:22]: Your PR was merged, this is no longer needed,

File: restapi/client.go [hunk 2]
```diff
@@ -107,3 +130,46 @@ func newMinioClient() (*minio.Client, error) {
...
+	if !ok {
+		return nil, fmt.Errorf("the provided url doesn't point to a S3 server")
+	}
+
+	return s3Client, nil
+}
+
+// newS3Config simply creates a new Config struct using the passed
+// parameters.
+func newS3Config(endpoint, accessKey, secretKey string, isSecure bool) *mc.Config {
```
Comment (dvaldivia) [inline restapi/client.go:159]: This is algo present in `mc` maybe we should make it public?

File: restapi/user_buckets_events.go [hunk 3]
```diff
@@ -130,3 +136,50 @@ func getListBucketEventsResponse(params user_api.ListBucketEventsParams) (*model
 	}
 	return listBucketsResponse, nil
 }
+
+// createBucketEvent calls mc AddNotificationConfig() to create a bucket nofication
+//
+// If notificationEvents is empty, by default will set [get, put, delete], else the provided
+// ones will be set.
+// this function follows same behavior as minio/mc for adding a bucket event
+func createBucketEvent(client MCS3Client, arn string, notificationEvents []models.NotificationEventType, prefix, suffix string, ignoreExisting bool) error {
```
Comment (dvaldivia) [inline restapi/user_buckets_events.go:145]: Should we make arn an `enum` ? it has a finite number of values

File: restapi/client.go (hunk 2 above)
Comment (cesnietor) [reply restapi/client.go:159]: not this is specific for us, the one used on mc doesn't work for us

File: restapi/user_buckets_events.go (hunk 3 above)
Comment (cesnietor) [reply restapi/user_buckets_events.go:145]: arn is just a string, what you men?

File: go.mod (hunk 1 above)
Comment (cesnietor) [reply go.mod]: done

File: restapi/user_buckets_events.go (hunk 3 above)
Comment (dvaldivia) [reply restapi/user_buckets_events.go:145]: that there's only like 10 possible values

**Pull request merged**